kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/engine/DefaultCommitCoordinatorClientHandler.java

```diff
@@ -0,0 +1,285 @@
+        Path path = new Path(logPath);
+        LogStore logStore = LogStoreProvider.getLogStore(hadoopConf, path.toUri().getScheme());
+        return convertCommitResponse(commitCoordinatorClient.commit(
+                logStore,
+                hadoopConf,
+                path,
+                tableConf,
+                commitVersion,
+                new Iterator<String>() {
+                    @Override
+                    public boolean hasNext() {
+                        return actions.hasNext();
+                    }
+
+                    @Override
+                    public String next() {
+                        return JsonUtils.rowToJson(actions.next());
+                    }
+                },
+                convertUpdatedActions(updatedActions)));
+    }
+
+    @Override
+    public GetCommitsResponse getCommits(
+            String tablePath,
+            Map<String, String> tableConf,
+            Long startVersion,
+            Long endVersion) {
+        return convertGetCommitsResponse(commitCoordinatorClient.getCommits(
+                new Path(tablePath),
+                tableConf,
+                startVersion,
+                endVersion));
+    }
+
+    @Override
+    public void backfillToVersion(
+            String logPath,
+            Map<String, String> tableConf,
+            long version,
+            Long lastKnownBackfilledVersion) {
+        Path path = new Path(logPath);
+        LogStore logStore = LogStoreProvider.getLogStore(hadoopConf, path.toUri().getScheme());
+        commitCoordinatorClient.backfillToVersion(
+                logStore,
+                hadoopConf,
+                path,
+                tableConf,
+                version,
+                lastKnownBackfilledVersion);
+    }
+
+    @Override
+    public Boolean semanticEquals(CommitCoordinatorClientHandler other) {
+        return other instanceof DefaultCommitCoordinatorClientHandler &&
+                commitCoordinatorClient.semanticEquals(
+                ((DefaultCommitCoordinatorClientHandler) other).getCommitCoordinatorClient());
+    }
+
+    public CommitCoordinatorClient getCommitCoordinatorClient() {
+        return commitCoordinatorClient;
+    }
+
+    private io.delta.storage.commit.UpdatedActions convertUpdatedActions(
+            UpdatedActions updatedActions) {
+        if (updatedActions == null) {
+            return null;
+        }
+        return new io.delta.storage.commit.UpdatedActions(
+                convertAbstractCommitInfo(updatedActions.getCommitInfo()),
+                convertAbstractMetadata(updatedActions.getNewMetadata()),
+                convertAbstractProtocol(updatedActions.getNewProtocol()),
+                convertAbstractMetadata(updatedActions.getOldMetadata()),
+                convertAbstractProtocol(updatedActions.getOldProtocol()));
+    }
+
+    private CommitResponse convertCommitResponse(io.delta.storage.commit.CommitResponse response) {
```

Comment:
In my opinion, these adapters should be moved a separate converter/adapter class and the function names should be of the form `toStorageUpdatedActions` and `toKernelAPIUpdatedActions`.